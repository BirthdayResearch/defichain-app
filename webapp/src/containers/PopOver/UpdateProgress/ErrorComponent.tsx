import React from 'react';
import { Button, ModalBody, ModalFooter } from 'reactstrap';
import { connect } from 'react-redux';
import { I18n } from 'react-redux-i18n';
import { closeUpdateApp } from '../reducer';
import openNewTab from '../../../utils/openNewTab';
import { SITE_DOWNLOADS } from '@defi_types/settings';
import { getSiteURL } from '../../../constants';

interface DownloadProgressComponentProps {
  isUpdateError: string;
  closeUpdateApp: () => void;
}

const DownloadProgressComponent = (props: DownloadProgressComponentProps) => {
  const { isUpdateError, closeUpdateApp } = props;
  const onDownloadClick = () => {
    openNewTab(`${getSiteURL()}${SITE_DOWNLOADS}`);
    closeUpdateApp();
  };
  return (
    <>
      <ModalBody>{isUpdateError}</ModalBody>
      <ModalFooter>
        <Button size='sm' color='link' onClick={closeUpdateApp}>
          {I18n.t('alerts.closeBtnLabel')}
        </Button>
        <Button size='sm' color='primary' onClick={onDownloadClick}>
          {I18n.t('alerts.download')}
        </Button>
      </ModalFooter>
    </>
  );
};

const mapStateToProps = ({ popover }) => ({
  isUpdateError: popover.isUpdateError,
});

const mapDispatchToProps = {
  closeUpdateApp,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DownloadProgressComponent);
