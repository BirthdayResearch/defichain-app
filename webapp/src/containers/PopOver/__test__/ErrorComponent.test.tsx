import React from 'react';
import { Provider } from 'react-redux';
import store from '../../../app/rootStore';
import Enzyme from '../../../utils/testUtils/enzyme';
import ErrorComponent from '../UpdateProgress/ErrorComponent';

describe('ErrorComponent', () => {
  it('should check for snapshot', () => {
    const wrapper = Enzyme.mount(
      <Provider store={store}>
        <ErrorComponent />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
