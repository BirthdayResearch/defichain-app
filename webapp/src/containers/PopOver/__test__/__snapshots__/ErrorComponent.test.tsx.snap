// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent should check for snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(DownloadProgressComponent)>
    <DownloadProgressComponent
      closeUpdateApp={[Function]}
      isUpdateError=""
    >
      <ModalBody
        tag="div"
      >
        <div
          className="modal-body"
        />
      </ModalBody>
      <ModalFooter
        tag="div"
      >
        <div
          className="modal-footer"
        >
          <Button
            color="link"
            onClick={[Function]}
            size="sm"
            tag="button"
          >
            <button
              aria-label={null}
              className="btn btn-link btn-sm"
              onClick={[Function]}
              type="button"
            >
              Download
            </button>
          </Button>
          <Button
            color="primary"
            onClick={[Function]}
            size="sm"
            tag="button"
          >
            <button
              aria-label={null}
              className="btn btn-primary btn-sm"
              onClick={[Function]}
              type="button"
            >
              Close
            </button>
          </Button>
        </div>
      </ModalFooter>
    </DownloadProgressComponent>
  </Connect(DownloadProgressComponent)>
</Provider>
`;
